common/config/config_parser.go

```diff
@@ -275,7 +273,18 @@ func SetBool(key string, val bool) {
 }
 
 func IsSet(key string) bool {
-	return viper.IsSet(key)
+	if viper.IsSet(key) {
```

Comment:
what are we doing here?

Reply:
first checking if its in config otherwise travel through the flag tree and see if its provided there